src/components/ListItem.jsx

```diff
@@ -1,5 +1,59 @@
 import './ListItem.css';
+import { useState, useEffect } from 'react';
+import { updateItem } from '../api';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const one_day_in_ms = 24 * 60 * 60 * 1000;
```

Comment:
Refactor (nitpick): The constant doesn't need to be in the component. It can live outside of it.

> timeElapsed being defined as a constant here means it's not going to change or react based on database changes or value changes, it's going to be set once on page load and never update. I found if I changed the database value to a date more than 24 hours ago I had to manually refresh the page to have the input box show unchecked

We're not using websockets or something similar, i.e. realtime, but you could have a function that polls every n minutes to see if 24 hours is up, and if so you could have a piece of state. The thing is you can do that in the `<ListItem />` component as the item is the prop passed in. You would need to do it in the `<App />` component as that's where all the data comes from. There could be other ways to do this, but at the moment with the where the data is passed in, that would probably make sense.

Reply:
Hey @nickytonline could we maybe hop on a zoom chat or something for a few minutes today? I'm not quite sure I'm understanding the requested changes. I think what you're asking is to refactor our logic in ListItem into App.jsx or List.jsx and to use the onSnapshot function. Is that correct? And this needs to be done in order to have the item uncheck themselves live on the page, rather than requiring a refresh to occur?